dateutil/tz/tz.py

```diff
@@ -904,6 +912,21 @@ def __repr__(self):
     TZFILES = []
     TZPATHS = []
 
+def tz_parserinfo():
+    if not hasattr(tz_parserinfo, 'info'):
+        tmplst = set()
+        for path in TZPATHS:
+            if not os.path.isdir(path):
+                continue
+
+            map(tmplst.add, ifilter(partial(re.match, '^[-\w][-+\w]*$'), os.listdir(path)))
+        from ..zoneinfo import initclasszone
+        info = initclasszone().parserinfo()
+        tmplst = tmplst.union(info[0])
+        tz_parserinfo.info = (tmplst, max(3, info[1]))
+
+    return tz_parserinfo.info
```

Comment:
@aschatten Can you explain what this is supposed to be? I get the sense that it's for identifying strings that might be time zones we know about, but what is the 2nd parameter for? The maximum number of segments it could be?